tests/integration/test_dss.py

```diff
@@ -130,26 +130,34 @@ def test_log_command(cleanup_after_initialize) -> None:
     assert "Starting gunicorn" in result.stderr
 
 
-@pytest.fixture(scope="module")
-def cleanup_after_initialize():
-    """Cleans up resources that might have been deployed by dss initialize.
-
-    Note that this is a white-box implementation - it depends on knowing what could be deployed and
-    explicitly removing those objects, rather than truly restoring the cluster to a previous state.
-    This could be leaky, depending on how `dss` is changed in future.  Hopefully though, by cleaning
-    up the Namespace, anything that might leak through will still be caught.
+def test_purge() -> None:
+    """
+    Tests that `purge` command removes all notebooks and DSS components.
     """
-    yield
-
-    k8s_resource_handler = KubernetesResourceHandler(
-        field_manager=FIELD_MANAGER,
-        labels=DSS_CLI_MANAGER_LABELS,
-        template_files=[],
-        context={},
-        resource_types={Deployment, Service, PersistentVolumeClaim, Namespace},
+    kubeconfig_file = "~/.kube/config"
+    kubeconfig = lightkube.KubeConfig.from_file(kubeconfig_file)
+    lightkube_client = lightkube.Client(kubeconfig)
+
+    # Run the purge command with the notebook name and kubeconfig file
+    result = subprocess.run(
+        [
+            "dss",
+            "purge",
+            "--kubeconfig",
+            kubeconfig_file,
+        ],
+        capture_output=True,
+        text=True,
+    )
+
+    # Check if the command executed successfully
+    assert result.returncode == 0
+    assert (
+        "Success: All DSS components and notebooks purged successfully from the Kubernetes cluster."
+        in result.stderr
     )
 
-    # Attempt to clean up anything that initialize might create
-    # Note that .delete() does not wait on the objects to be successfully deleted, so repeating
-    # the tests quickly can still cause an issue
-    k8s_resource_handler.delete()
```

Comment:
it's a bit tricky testing the removal in the integration tests, removing the cleanup fixture means that if something goes wrong with the purge command then the test resources will stay there. Maybe we can keep the cleanup fixture as a fail-safe. wdyt?

Reply:
Good point actually